packages/babel-runtime/scripts/build-dist.js

```diff
@@ -82,23 +93,40 @@ function buildRuntimeRewritePlugin(relativePath, helperName) {
   };
 }
 
-function buildHelper(helperName) {
+function buildHelper(helperName, modules, useBuiltIns) {
+  const helper = helpers.get(helperName)
+  // avoid an unneccessary TDZ in the easy case
+  if (helper.type === "FunctionExpression") {
+    helper.type = "FunctionDeclaration"
+  }
   var tree = t.program([
-    t.exportDefaultDeclaration(helpers.get(helperName))
+    t.exportDefaultDeclaration(helper)
   ]);
 
+  const transformOpts = makeTransformOpts(modules, useBuiltIns)
+
+  const relative = useBuiltIns ? "../.." : ".."
+
   return babel.transformFromAst(tree, null, {
     presets: transformOpts.presets,
-    plugins: transformOpts.plugins.concat([buildRuntimeRewritePlugin("..", helperName)])
+    plugins: transformOpts.plugins.concat([buildRuntimeRewritePlugin(modules === false ? `../${relative}` : relative, helperName)])
   }).code;
 }
 
-helpers.list.forEach(function (helperName) {
-  writeFile("helpers/" + helperName + ".js", buildHelper(helperName));
+for (const modules of ["commonjs", false]) {
+  for (const builtin of [false, true]) {
+    const dirname = `helpers/${builtin ? 'builtin/' : ''}${!modules ? 'es6/' : ''}`
 
-  // compat
-  var helperAlias = kebabCase(helperName);
-  var content = "module.exports = require(\"./" + helperName + ".js\");";
-  writeFile("helpers/_" + helperAlias + ".js", content);
-  if (helperAlias !== helperName) writeFile("helpers/" + helperAlias + ".js", content);
-});
+    for (const helperName of helpers.list) {
+      writeFile(`${dirname}${helperName}.js`, buildHelper(helperName, modules, builtin));
+
+      // compat
+      var helperAlias = kebabCase(helperName);
+      var content = !modules
+        ? `export { default } from \"./${helperName}.js\";`
+        : "module.exports = require(\"./" + helperName + ".js\");";
+      writeFile(`${dirname}_${helperAlias}.js`, content);
+      if (helperAlias !== helperName) writeFile(`${dirname}${helperAlias}.js`, content);
```

Comment:
We should really kill all these aliases in 7.x. If you're up for it, maybe a PR another this one? Let's keep this separate so if we want to backport it is easier.